File: 1/1_1.go
package main

import (
	"os"
	"fmt"
	"bufio"
	"strconv"
)

func main() {
	var count int = 0
	var last int = -1 // FIXME: learn a more idiomatic golang way to do this

	file, err := os.Open("data.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		n, _ := strconv.Atoi(scanner.Text())

		if (last != -1 && n > last) {
			count++
		}

		last = n
	}

	fmt.Println(count)
}

File: 1/1_1.py
#!/usr/bin/python3
from pprint import pprint

count = 0
last = None

with open('data.txt') as file:
    for line in file:
        n = int(line.rstrip())
        if last is not None and n > last:
            count += 1;
        last = n

pprint(count)
